src/test/kotlin/uk/gov/dluhc/printapi/messaging/SendApplicationToPrintMessageListenerIntegrationTest.kt

```diff
@@ -4,28 +4,97 @@ import ch.qos.logback.classic.Level
 
     @Test
     fun `should process message received on queue`() {
         // Given
-        val payload = buildSendApplicationToPrintMessage()
+        val ero = buildElectoralRegistrationOfficeResponse()
+        val localAuthority = ero.localAuthorities[1]
+        val gssCode = localAuthority.gssCode!!
+        val payload = buildSendApplicationToPrintMessage(gssCode = gssCode)
+        wireMockService.stubEroManagementGetEro(ero, gssCode)
+        val expected = with(payload) {
+            PrintDetails(
+                id = UUID.randomUUID(),
+                requestId = aValidRequestId(),
+                sourceReference = sourceReference,
+                applicationReference = applicationReference,
+                sourceType = SourceType.VOTER_CARD,
+                vacNumber = aValidVacNumber(),
+                requestDateTime = requestDateTime,
+                firstName = firstName,
+                middleNames = middleNames,
+                surname = surname,
+                certificateLanguage = CertificateLanguage.EN,
+                photoLocation = photoLocation,
+                delivery = with(delivery) {
+                    CertificateDelivery(
+                        addressee = addressee,
+                        address = with(address) {
+                            Address(
+                                street = street,
+                                postcode = postcode,
+                                property = property,
+                                locality = locality,
+                                town = town,
+                                area = area,
+                                uprn = uprn
+                            )
+                        },
+                        deliveryClass = uk.gov.dluhc.printapi.database.entity.DeliveryClass.STANDARD
+                    )
+                },
+                gssCode = gssCode,
+                issuingAuthority = localAuthority.name,
+                issueDate = LocalDate.now(),
+                eroEnglish = with(ero) {
+                    ElectoralRegistrationOffice(
+                        name = name,
+                        phoneNumber = null,
+                        emailAddress = null,
+                        website = null,
+                        address = null
+                    )
+                },
+                eroWelsh = null
+            )
+        }
 
         // When
         sqsMessagingTemplate.convertAndSend(sendApplicationToPrintQueueName, payload)
 
         // Then
         await.atMost(5, SECONDS).untilAsserted {
-            assertThat(
-                TestLogAppender.hasLog(
-                    "Sending application [${payload.sourceReference}] to print", Level.INFO
-                )
-            ).isTrue()
+            wireMockService.verifyEroManagementGetEro(gssCode)
+            val response = dynamoDbClient.scan(
```

Comment:
Is there any mileage here to use the repository rather than the dynamo client?
(I don't know what the repository class has in it yet wrt query/read methods - if there are no suitable methods in the repository yet then what you've done here is probably right. But if there are suitable methods would that make this test a little tidier?)

Reply:
Hi, @NathanRussellValtech ,
The repository currently only contains getting items by the primary key, hence needed to use the dynamodb client.

Reply:
I thought that might be the case; but thought I'd ask anyway 😁 
Thats cool 👍